hub/cli/auth.py

```diff
@@ -19,32 +19,41 @@
 @click.option("--password", "-p", default=None, help="Your Activeloop Password")
 def login(username: str, password: str):
     """Log in to Activeloop"""
-    if not username:
-        click.echo("Login to Activeloop using your credentials.")
-        click.echo(
-            "If you don't have an account, register by using the 'activeloop register' command or by going to "
-            f"{HUB_REST_ENDPOINT}/register."
-        )
-        username = click.prompt("Username")
-        password = click.prompt("Password", hide_input=True)
-    if not password:
-        password = click.prompt(
-            f"Please enter password for user {username}", hide_input=True
-        )
-    username = username.strip()
-    password = password.strip()
-    try:
-        client = HubBackendClient()
-        token = client.request_auth_token(username, password)
-        write_token(token)
-        click.echo("Successfully logged in to Activeloop.")
-        reporting_config = get_reporting_config()
-        if reporting_config.get("username") != username:
-            save_reporting_config(True, username=username)
-    except AuthenticationException:
-        raise SystemExit("Login failed. Check username and password.")
-    except Exception as e:
-        raise SystemExit(f"Unable to login: {e}")
+    chances: int = 3
+    while chances:
+        if not username:
+            click.echo("Login to Activeloop using your credentials.")
+            click.echo(
+                "If you don't have an account, register by using the 'activeloop register' command or by going to "
+                f"{HUB_REST_ENDPOINT}/register."
+            )
+            username = click.prompt("Username")
+            password = click.prompt("Password", hide_input=True)
+        if not password:
+            password = click.prompt(
+                f"Please enter password for user {username}", hide_input=True
+            )
+        username = username.strip()
+        password = password.strip()
+        try:
+            client = HubBackendClient()
+            token = client.request_auth_token(username, password)
+            write_token(token)
+            click.echo("Successfully logged in to Activeloop.")
+            reporting_config = get_reporting_config()
+            if reporting_config.get("username") != username:
+                save_reporting_config(True, username=username)
+            break
+        except AuthenticationException:
+            print("Login failed. Check username and password.")
+            chances -= 1
+            username = ""
+            password = ""
+        except Exception as e:
```

Comment:
We shouldn't try 3 times if there's an exception coming from the backend. The user can't do anything about it and providing creds 3 times might be annoying.
Let's instead reword this print statement to say something like f"Encountered an error {e}, please try again later." and exit.

Reply:
@gaurav879 this still hasn't been fixed, for generic exceptions we shouldn't try 3 times, but exit at the first failure instead, after printing the error